service/controller/chart/v1/resource/release/delete.go

```diff
@@ -54,7 +54,7 @@ func (r *Resource) newDeleteChange(ctx context.Context, obj, currentState, desir
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding out if the %#q release has to be deleted", desiredReleaseState.Name))
 
-	if !isEmpty(currentReleaseState) && equals(currentReleaseState, desiredReleaseState) {
+	if !isEmpty(currentReleaseState) && currentReleaseState.Name == desiredReleaseState.Name {
```

Comment:
Now if the chart CR has been deleted the desired ReleaseState struct will only have the Name set.